geopandas/array.py

```diff
@@ -1224,23 +1224,27 @@ def _formatter(self, boxed=False):
 
             precision = geopandas.options.display_precision
             if precision is None:
-                # dummy heuristic based on 10 first geometries that should
-                # work in most cases
-                with warnings.catch_warnings():
-                    warnings.simplefilter("ignore", category=RuntimeWarning)
-                    xmin, ymin, xmax, ymax = self[~self.isna()][:10].total_bounds
-                if (
-                    (-180 <= xmin <= 180)
-                    and (-180 <= xmax <= 180)
-                    and (-90 <= ymin <= 90)
-                    and (-90 <= ymax <= 90)
-                ):
-                    # geographic coordinates
-                    precision = 5
-                else:
-                    # typically projected coordinates
-                    # (in case of unit meter: mm precision)
+                if self.crs and self.crs.is_projected:
                     precision = 3
```

Comment:
```suggestion
                if self.crs
                    if self.crs.is_projected:
                        precision = 3
                    else:
                        precision = 5
```

No need to go through fallback for a known geographic projection.